src/main/java/org/kainos/groupA/dao/UserDao.java

```diff
@@ -43,4 +64,43 @@ public int createUser(User user, Connection c) throws SQLException {
         }
         return userId;
     }
+
+    public Map<String, String> loginUser(LoginUser loginUser, Connection c) throws SQLException, InvalidUserException, JoseException {
+        try {
+            String checkLoginQuery = "SELECT email, role FROM User WHERE email=? AND password=?";
+            PreparedStatement preparedSt = c.prepareStatement(checkLoginQuery, Statement.RETURN_GENERATED_KEYS);
+            preparedSt.setString(1, loginUser.getEmail());
+            preparedSt.setString(2, loginUser.getPassword());
+            ResultSet rs = preparedSt.executeQuery();
+            if(rs.next()) {
+                final JwtClaims claims = new JwtClaims();
+                claims.setSubject(rs.getString(1));
+                claims.setExpirationTimeMinutesInTheFuture(43200);
+
+                final JsonWebSignature jws = new JsonWebSignature();
+                jws.setPayload(claims.toJson());
+                jws.setAlgorithmHeaderValue(HMAC_SHA256);
+                jws.setKey(new HmacKey(tokenSecret));
+                Map<String, String> map = new HashMap<>();
+                map.put("token", jws.getCompactSerialization());
+                map.put("role", rs.getString(2));
```

Comment:
the role should be stored in the token, you don't need to return a map here, you should just return an encrypted JWT token as a string